### ISSUE TITLE
Shift ranged bonuses from mods to guns and skills.

### ISSUE THREAD

**Issue by kevingranade** [CleverRaven/Cataclysm-DDA#24429]
### Expected behavior
Ability to hit with ranged weapons is dominated by skills, stats, and firearm properties.
A highly skilled player with a good gun can achieve maximum firearm performance.
### Actual behavior
Mods provide a disproportionate amount of gun accuracy.
It's impossible to reach peak gun accuracy without a heavily modded gun.

Proposal is that we shift the majority of gun accuracy from mods into the underlying guns, de-emphasizing the importance of gun mods for basic gun performance.

This is a continuation of #22211

**Comment by darktoes** [comment]
I agree that the majority should come from guns themselves, but I do think that reaching the absolute maximum accuracy should require some mods. Extended barrel and ergonomic grip (to a lesser extent) in particular. That said, some high accuracy weapons like sniper rifles probably shouldn't have a slot for an extended barrel, or should spawn with it already applied. That would give players the option of paring down sniper rifles into a short and powerful, but inefficient, "hand" cannon.

**Comment by kevingranade** [comment]
Extended barrel is an example of a mod that should mostly go away, its utter nonsense that a single mod or kit can extend the barrel on arbitrary rifles.
If a particular gun has swappable barrels, that should be represented by a gun-specific mod, not a generic one.

**Comment by darktoes** [comment]
If we're going to be removing all of the unrealistic gun mods, should we also get rid of the gunsmith kit in it's entirety? Other than crafting recipes and battery efficiency, does it serve any other purpose over the firearm repair kit?

**Comment by Firestorm01X2** [comment]
Wow, wow, wow.

I recommend to do just nerfing without throwing away things. It is easier.

**Comment by Firestorm01X2** [comment]
I have to note that there is one thing that prevent reaching maximum firearm performance - it is lack of specilized sight or scope. Sight dispersion without scope is +30, but suprisingly it is lot. Without it you can't even get close to 30 tiles of good hit range.

**Comment by Firestorm01X2** [comment]
@kevingranade 
>A highly skilled player with a good gun can achieve maximum firearm performance.

What is maximum firearm performance. I propose o determine it as  goog hit  range=  maximum gun range

**Comment by Firestorm01X2** [comment]
Ok. Let's get average 8 stat characted and non modded m4a1
Now
skills | >50% good
-- | -- 
Novice pistol | 5 |

**Comment by kevingranade** [comment]
> should we also get rid of the gunsmith kit

I haven't looked into it, if all of its uses get eliminated, then yes.

> That way the extended barrel isn't a kit at all, and you could customize material costs, time requirements, etc.

The problem with this approach is a gun barrel is something the survivor can't craft.

> Long barrel is too magic with -90 dispersion?

It's not a balance issue, the mere fact that you can have a generic barrel mod is the problem.  If we replace "long barrel kit" with "Remington 700 extended barrel kit", then adjusting dispersion a bit is fine.

> I have to note that there is one thing that hugely prevent reaching maximum firearm performance - it is lack of specilized sight or scope.

Good point, scopes are likely going to be an exception to this since they DO extend effective range of certain guns a great deal.

> What is maximum firearm performance?

This issue doesn't propose we change maximum effective range, it's just moving numbers from one set of items to another.  To be concrete, this is one of the test setups that's clearly a problem:
```
SECTION( "an expert shooter with a heavily modded auto shotgun" ) {
    arm_shooter( shooter, "abzats", { "holo_sight", "light_grip", "tuned_mechanism", "barrel_rifled" } );
    test_shooting_scenario( shooter, 18, 24, 124 );
    test_fast_shooting( shooter, 60, 0.5 );
}
```
This means that while the stated performance is achievable, the player has to track down all those random gunmods to make it happen, it *should* be:
```
SECTION( "an expert shooter with a heavily modded auto shotgun" ) {
    arm_shooter( shooter, "abzats", { } );
    test_shooting_scenario( shooter, 18, 24, 124 );
    test_fast_shooting( shooter, 60, 0.5 );
}
```

So the plan is basically:
Remove all non-sight gunmods from accuracy tests.
Either nerf or remove gunmods as appropriate based on whether they make sense (basically all barrel mods should go away).
Reevaluate which sights are applied at each level to make them appropriate for the level of advancement.
Rebalance underlying guns and accuracy code to make tests pass again.

**Comment by Firestorm01X2** [comment]
@kevingranade 
But what about ranges istelf. For example noew competent rifle shooter can achive 30 tiles of effective range.

Should we use same ranges?

**Comment by Firestorm01X2** [comment]
>Remove all non-sight gunmods from accuracy tests.

Ok. Got it.

**Comment by Night-Pryanik** [comment]
Closed after #24479.
